crates/deltalake-core/src/table/mod.rs

```diff
@@ -637,17 +638,31 @@ impl DeltaTable {
         &mut self,
         version: i64,
     ) -> Result<i64, DeltaTableError> {
+        // Use a cached timestamp info if available
         match self.version_timestamp.get(&version) {
             Some(ts) => Ok(*ts),
             None => {
-                let meta = self
-                    .object_store()
-                    .head(&commit_uri_from_version(version))
-                    .await?;
-                let ts = meta.last_modified.timestamp_millis();
+                // Load the version specified
+                if self.version() != version {
+                    self.load_version(version).await?;
+                }
+                let timestamp: Option<i64> = if !self.state.commit_infos().is_empty() {
+                    self.state.commit_infos().last().unwrap().timestamp
+                } else {
+                    None
+                };
```

Comment:
This looks obsolete because `self.version_timestamp` is a HashMap of the table versions and their timestamps which to my understanding are coming from the transaction log.

Reply:
Ah it seems we are not initializing any timestamp:version in the hashmap, only when it's not there its cached.